Mathlib/Combinatorics/Enumerative/Partition/Basic.lean

```diff
@@ -117,6 +122,38 @@ def ofSymShapeEquiv (μ : Partition n) (e : σ ≃ τ) :
   left_inv := by intro x; simp
   right_inv := by intro x; simp
 
+/-- Convert a `Partition n` to a member of `(Finset.Icc 1 n).finsuppAntidiag n`
+(see `Nat.Partition.toFinsuppAntidiag_mem_finsuppAntidiag` for the proof).
+`p.toFinsuppAntidiag i` is defined as `i` times the number of occurrence of `i` in `p`. -/
+def toFinsuppAntidiag {n : ℕ} (p : Partition n) : ℕ →₀ ℕ where
+  toFun m := p.parts.count m * m
+  support := p.parts.toFinset
+  mem_support_toFun m := by
+    suffices m ∈ p.parts → m ≠ 0 by simpa
+    grind [→ parts_pos]
```

Comment:
I think it'd be reasonable to mark `parts_pos` as `grind →`

Reply:
How do I actually do that? `parts_pos` is a field of Partition. Trying adding `@[grind →]` there gives me `Invalid attribute: Attributes cannot be added to fields`
Should I create a standalone theorem `parts_pos` for this, and rename the field to `parts_pos'`?

Reply:
Fixed. Thanks for the suggestion!